be/src/olap/generic_iterators.cpp

```diff
@@ -266,9 +266,17 @@ class MergeIterator : public RowwiseIterator {
             // here we sort segment id in reverse order, because of the row order in AGG_KEYS
             // dose no matter, but in UNIQUE_KEYS table we only read the latest is one, so we
             // return the row in reverse order of segment id
-            bool result = res == 0 ? lhs->data_id() < rhs->data_id() : res < 0;
-            if (_is_unique) {
-                result ? lhs->set_skip(true) : rhs->set_skip(true);
+            bool result = false;
```

Comment:
Please add some comments.
The following method may be clearer.
```
 if (_is_unique) {
     bool result = res == 0 ? lhs->data_id() < rhs->data_id() : res < 0;
     result ? lhs->set_skip(true) : rhs->set_skip(true);
     return result;
 }     
 return lhs->data_id() > rhs->data_id();
```

Reply:
thanks, it is better